pkg/propertychecker/azure/checker.go

```diff
@@ -0,0 +1,167 @@
+//
+//Copyright (c) Microsoft Corporation.
+//Licensed under the MIT license.
+
+// Package azure provides property checkers for Azure-specific cluster requirements.
+// It checks whether the cluster can meet the requirement defined by the property selector.
+package azure
```

Comment:
just realized that this is not part of the Azure provider? I wonder why? This is only going to be used by the azure provider, right?

Reply:
property checker is a interface to check if the cluster meets the property requirement. The current azure one is not generic enough and it can be used by other properties if we have, which we can refactor later. It won't be used by the provider and will be used by the clusteraffinity plugin instead.

While the property provider is to collect and store the property as part of the member cluster CR.